monitoring/scripts/central-monitor-setup.ps1

```diff
@@ -0,0 +1,71 @@
+
+get-content ../ingress/aadpodidentity-template.txt | ForEach-Object {$_.replace("`${name}","gpitfutures-monitor-aad-id")} | ForEach-Object {$_.replace("`${idPath}","$aadIdPath")} | ForEach-Object {$_.replace("`${clientID}","$aadClientId")} > ./scripts/ingress/monitor-aadpodidentity.yaml
+get-content ../ingress/aadpodidentitybinding-template.txt | ForEach-Object {$_.replace("`${name}","gpitfutures-monitor-aad-id")} | ForEach-Object {$_.replace("`${clientID}","$aadClientId")} > ./scripts/ingress/monitor-aadpodidentitybinding.yaml
+
+kubectl apply -f ./scripts/ingress/monitor-aadpodidentity.yaml -n ingress-appgw
+kubectl apply -f ./scripts/ingress/monitor-aadpodidentitybinding.yaml -n ingress-appgw
+
+az role assignment create --role "Managed Identity Operator" --assignee $aksSpId --scope $aadIdPath
+
+$agicshared="true"
+get-content ../ingress/helm-agic-config-template.txt | ForEach-Object {$_.replace("`${subId}","$subId")} | ForEach-Object {$_.replace("`${appgwRg}","$appgwRgName")} | ForEach-Object {$_.replace("`${appgwName}","gpitfutures-monitor-appgw")} | ForEach-Object {$_.replace("`${aadId}","$aadIdPath")} | ForEach-Object {$_.replace("`${clientId}","$aadClientId")} | ForEach-Object {$_.replace("`${apiAddress}","$aksFqdn")} | ForEach-Object {$_.replace("`${shared}","$agicshared")} > ./scripts/ingress/monitor-helm-config.yaml
+
+helm repo add application-gateway-kubernetes-ingress https://appgwingress.blob.core.windows.net/ingress-azure-helm-package/
+helm repo update
+helm install ingress-azure -f ./scripts/ingress/monitor-helm-config.yaml application-gateway-kubernetes-ingress/ingress-azure -n ingress-appgw --version 1.2.0-rc3
+kubectl apply -f https://raw.githubusercontent.com/Azure/application-gateway-kubernetes-ingress/ae695ef9bd05c8b708cedf6ff545595d0b7022dc/crds/AzureIngressProhibitedTarget.yaml
+
+$agicCrdURL="rancher.dynamic.buyingcatalogue.digital.nhs.uk"
+$agicCrdName="rancher"
+get-content ../ingress/azureingressprohibitedtarget.txt | ForEach-Object {$_.replace("`${PTname}","$agicCrdName")} | ForEach-Object {$_.replace("`${URL}","$agicCrdURL")} > ./scripts/ingress/rancher-monitor-azureingressprohibitedtarget.yaml
+kubectl apply -f ./scripts/ingress/rancher-monitor-azureingressprohibitedtarget.yaml -n ingress-appgw
+
+helm repo add rancher-stable https://releases.rancher.com/server-charts/stable
+helm repo update
+kubectl create namespace cattle-system
+helm install rancher rancher-stable/rancher --namespace cattle-system --set hostname=rancher.dynamic.buyingcatalogue.digital.nhs.uk --set tls=external
+
+start-sleep 120 # to allow rancher pods to deploy
+
+#ipArr=()
+#        ipArr+=$(kubectl get pods -l app=rancher -n cattle-system -o wide | while read line ; do while read a b c d e ip g h i; do echo $ip ; done ; done)
+#        az network application-gateway address-pool update -g gpitfutures-monitor-rg-appgw --gateway-name gpitfutures-monitor-appgw -n rancher --servers $ipArr  
```

Comment:
Trailing new line. 😃

Reply:
Replaced by PS1 version and #'d back in - thanks!